src/test/java/TestLinks.java

```diff
@@ -0,0 +1,69 @@
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Created by pavel on 22/11/2017.
+ */
+public class TestLinks {
+
+    List<URL> list;
+    
+    @Before
+    public void createListLinks() {
+        Links<URL> links = new DecoratorForURL (new DecoratorForString("http://www.tutorialspoint.com/"));
```

Comment:
@PavelNejinskiy lets introduce another implementataion of `Links` that will encapsulate those two. This implementation will be accepting `url` as input argument.